contrib/udfs/src/main/java/org/apache/drill/exec/udfs/NearestDateUtils.java

```diff
@@ -0,0 +1,111 @@
+package org.apache.drill.exec.udfs;
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.time.temporal.ChronoUnit;
+
+public class NearestDateUtils {
+  public final static java.time.LocalDateTime getDate(java.time.LocalDateTime d, String interval) {
```

Comment:
1. Please add java doc.
2. Please consider using enum for interval values.
3. Please make sure that interval check is case insensitive.

Reply:
Fixed.